web/views.py

```
@@ -7103,3 +7128,112 @@ def users_list(request: HttpRequest) -> HttpResponse:
+        # After processing the vote, redirect back to the topic page
+        return redirect("forum_topic", category_slug=reply.topic.category.slug, topic_id=reply.topic.id)
+
+    except ForumReply.DoesNotExist:
+        messages.error(request, "Reply not found")
+        return redirect("forum_categories")
+
+    except ForumReply.DoesNotExist:
+        return JsonResponse({"error": "Reply not found"}, status=404)
+
+
+def topic_detail(request, pk):
```

Ruff (0.8.2)

Missing return type annotation for public function `topic_detail`

(ANN201)